internal/api/service/health.go

```diff
@@ -4,10 +4,37 @@ import (
 	"time"
 
 	"github.com/base-org/pessimism/internal/api/models"
+	"github.com/base-org/pessimism/internal/logging"
+	"go.uber.org/zap"
 )
 
 // CheckHealth ... Returns health check for server
-// NOTE - As of now, this is a hardcoded function
 func (svc *PessimismService) CheckHealth() *models.HealthCheck {
-	return &models.HealthCheck{Timestamp: time.Now(), Healthy: true}
+	healthCheck := &models.NodeConnectionStatus{}
+
+	healthCheck.IsL1Healthy = svc.CheckETHRPCHealth(svc.cfg.L1RpcEndpoint)
+	healthCheck.IsL2Healthy = svc.CheckETHRPCHealth(svc.cfg.L2RpcEndpoint)
+
+	healthy := healthCheck.IsL1Healthy && healthCheck.IsL2Healthy
+
+	return &models.HealthCheck{Timestamp: time.Now(), Healthy: healthy}
+}
+
+func (svc *PessimismService) CheckETHRPCHealth(url string) bool {
+	logger := logging.WithContext(svc.ctx)
+
+	err := svc.ethClient.DialContext(svc.ctx, url)
+	if err != nil {
+		logger.Error("error conntecting to %s", zap.String("url", url))
+		return false
+	}
+
+	_, err = svc.ethClient.HeaderByNumber(svc.ctx, nil)
```

Comment:
I almost wonder if just being able to dial a connection would be sufficient enough. Grabbing the block header may be erroneous given that we aren't actually using it for anything. The only thing that comes to mind is validating that the chain is continuously producing new blocks. That would however be out of scope for this health checking endpoint.

Reply:
Tested this locally by removing the call to getHeaderByNumber and running the app then turning off wifi and it always returns true when we just check that the DialContext call doesn't fail. Not sure why this is happening, but this is why I initially added a secondary call to get data from the connection

Reply:
Tested this locally by removing the call to getHeaderByNumber and running the app then turning off wifi and it always returns true when we just check that the DialContext call doesn't fail. Not sure why this is happening, but this is why I initially added a secondary call to get data from the connection

Reply:
Ah you are correct, looking at the rpc logic in [geth](https://github.com/ethereum/go-ethereum/blob/master/rpc/client.go#L174-L184), this method just appears to be initializing metadata and performs no actual node RPC queries.

Reply:
Resolving in the meantime!